Game/Scripts/PlayerAttackScript.cpp

```diff
@@ -403,50 +403,70 @@ void PlayerAttackScript::InitJumpAttack()
 
 void PlayerAttackScript::UpdateJumpAttack()
 {
-	bool landed = false;
-	//if (isMelee) landing is player grounded if not then is the projectile detection for the moment I only put this
-
-	if (isMelee)
+	bool successfulAttack = false;
+	if (isMelee) 
 	{
-		landed = playerManager->IsGrounded();
+		successfulAttack = playerManager->IsGrounded();
 	}
 	else
 	{
-		AXO_TODO("Add Allura Checks")
-		landed = true;
+		successfulAttack = jumpFinisherScript->GetBulletHitTheFloor();
 	}
 
-	if (landed)
+	if (successfulAttack)
 	{
 		animation->SetParameter("IsJumpAttacking", false);
-		if (currentAttack == AttackType::JUMPNORMAL)
+		if (!comboSystem->NextIsSpecialAttack())
 		{
 			EndJumpNormalAttack();
 		}
 		else
 		{
 			EndJumpFinisherAttack();
 		}
+
+		if (!isMelee)
+		{
+			jumpFinisherScript->SetBulletHitTheFloor(false);
+		}
 		currentAttack = AttackType::NONE;
 	}
 }
 
 void PlayerAttackScript::EndJumpNormalAttack()
 {
 	jumpFinisherScript->VisualLandingEffect();
-	if (enemyDetection->AreAnyEnemiesInTheArea())
+
+	std::vector<ComponentRigidBody*> enemiesToHit = enemyDetection->GetEnemiesInTheArea();
+	if (!isMelee)
+	{
+		enemiesToHit.clear();
+		enemiesToHit.reserve(enemyDetection->GetEnemiesInTheArea().size());
+		enemyDetection->FilterEnemiesByDistance(6.5f, enemiesToHit); // 6.5f like the size of Bix jump attack
+	}
```

Comment:
Minor thing, but we could save on a function call (not sure how expensive `GetEnemiesInTheArea` is so not sure of the impact either).
```suggestion
	std::vector<ComponentRigidBody*> enemiesToHit;
	if (isMelee)
	{
			enemiesToHit = enemyDetection->GetEnemiesInTheArea();
	}
	else
	{
		enemiesToHit.reserve(enemyDetection->GetEnemiesInTheArea().size());
		enemyDetection->FilterEnemiesByDistance(6.5f, enemiesToHit); // 6.5f like the size of Bix jump attack
	}
```

Reply:
Well seen!